chain/chain/src/resharding.rs

```diff
@@ -179,6 +181,18 @@ fn set_flat_storage_state(
     Ok(())
 }
 
+/// Helper function to read the value from flat storage.
+/// It either returns the inlined value or reads ref value from the storage.
+fn read_flat_state_value(
+    trie_storage: &TrieDBStorage,
+    flat_state_value: FlatStateValue,
+) -> Vec<u8> {
+    match flat_state_value {
+        FlatStateValue::Ref(val) => trie_storage.retrieve_raw_bytes(&val.hash).unwrap().to_vec(),
+        FlatStateValue::Inlined(val) => val,
+    }
+}
+
```

Comment:
@robin-near, does this have anything to do with what you were fixing for in memory trie?